test/HarpoonBattleTest.cpp

```diff
@@ -0,0 +1,167 @@
+
+   EXPECT_TRUE(data == "<DATA>");
+   EXPECT_TRUE(done == "<DONE>");
+   EXPECT_TRUE(error == "<ERROR>");
+   EXPECT_TRUE(end == "<END>");
+
+}
+
+TEST_F(HarpoonBattleTest, EnumToString_Converted) {
+   auto data = HarpoonBattle::EnumToString(static_cast<HarpoonBattle::ReceivedType>(KrakenBattle::SendType::Data));
+   auto done = HarpoonBattle::EnumToString(static_cast<HarpoonBattle::ReceivedType>(KrakenBattle::SendType::Done));
+   auto error = HarpoonBattle::EnumToString(static_cast<HarpoonBattle::ReceivedType>(KrakenBattle::SendType::Error));
+   auto end = HarpoonBattle::EnumToString(static_cast<HarpoonBattle::ReceivedType>(KrakenBattle::SendType::End));
+
+   EXPECT_TRUE(data == "<DATA>");
+   EXPECT_TRUE(done == "<DONE>");
+   EXPECT_TRUE(error == "<ERROR>");
+   EXPECT_TRUE(end == "<END>");
+}
+
+
+// Disabled since it will trigger a fatal shutdown of the logger
+TEST_F(HarpoonBattleTest, EnumToString_Undefined) {
+   Death::SetupExitHandler();
+   Death::ClearExits();
+   EXPECT_FALSE(Death::WasKilled());
+   auto wrong = HarpoonBattle::EnumToString(static_cast<HarpoonBattle::ReceivedType>(1000));
+   auto found = wrong.find("<ERROR>");
+   EXPECT_TRUE(found != std::string::npos) << wrong;
+   EXPECT_TRUE(Death::WasKilled());
+
+   auto foundDeath = Death::Message().find("Unknown ReceivedType:");
+   EXPECT_TRUE(foundDeath != std::string::npos) << Death::Message();
+}
+
+TEST_F(HarpoonBattleTest, StringToEnum) {
+   auto data = HarpoonBattle::StringToEnum("<DATA>");
+   auto done = HarpoonBattle::StringToEnum("<DONE>");
+   auto error = HarpoonBattle::StringToEnum("<ERROR>");
+   auto end = HarpoonBattle::StringToEnum("<END>");
+   auto wrong = HarpoonBattle::StringToEnum("<Hello World>");
+
+   using namespace HarpoonBattle;
+   EXPECT_TRUE(data == ReceivedType::Data);
+   EXPECT_TRUE(done == ReceivedType::Done);
+   EXPECT_TRUE(error == ReceivedType::Error);
+   EXPECT_TRUE(end == ReceivedType::End);
+   EXPECT_TRUE(wrong == ReceivedType::Error);
+}
+
+
+
+TEST_F(HarpoonBattleTest, ChunksExtracted_DATA) {
+   const std::string uuid = "some-uuid";
+   auto type = KrakenBattle::SendType::Data;
+   auto data = KrakenIntegrationHelper::GetRandomData(1024);
+   auto error = std::string("no error");
+   auto merged = KrakenBattle::MergeData(uuid, type, data, error);
+
+   auto extracted = HarpoonBattle::ExtractToParts(merged);
+   EXPECT_EQ(std::get<HarpoonBattle::IndexOfSession>(extracted), uuid);
+
+   auto extractedType = std::get<HarpoonBattle::IndexOfReceivedType>(extracted);
+   EXPECT_EQ(extractedType, HarpoonBattle::ReceivedType::Data);
+
+   auto extractedChunk = std::get<HarpoonBattle::IndexOfChunk>(extracted);
+   auto dataAsString = KrakenIntegrationHelper::vectorToString(data);
+   auto extractedChunkAsString = KrakenIntegrationHelper::vectorToString(extractedChunk);
+   EXPECT_TRUE(dataAsString == extractedChunkAsString) << "data: " << dataAsString.size() << ",  extracted: " << extractedChunkAsString.size();
```

Comment:
I understood you meant something else... That all data including session and state would be all zeroes.

I'll verify it can handle zeroes or else I will use a vector comparison instead